sdks/python/apache_beam/runners/worker/sdk_worker_main.py

```diff
@@ -202,7 +202,7 @@ def _get_worker_count(pipeline_options):
           re.match(r'worker_threads=(?P<worker_threads>.*)',
                    experiment).group('worker_threads'))
 
-  return 12
+  return 10000
```

Comment:
Do we know why 10K is a good number? Would not this practically accept all work?

How did we come up with 12 in the first place. (@angoenka may have information)

Reply:
I choose a value where it is effectively unbounded.

Reply:
Do we have any benchmarks to look at and see what is the impact of this?

My underlying question is what does this imply for python streaming running with the java runner harness?

Reply:
It means that we will now schedule them to be executed while before we would block (which could lead to deadlock for specific pipeline shapes) where the execution of one bundle relies on the execution of another bundle.

Reply:
OK. That actually sounds good, could fix some stuckness issues.

I am not very clear on whether this will impact performance or not. We can at least just watch the changes after the merge. (cc: @angoenka and @y1chi as an FYI.)

Reply:
> Do we know why 10K is a good number? Would not this practically accept all work?
> 
> How did we come up with 12 in the first place. (@angoenka may have information)

It was just some basic experimentation where for the job which I was experimenting on did not show a lot of benefit beyond 12 threads.

For TFX pipelines, we recommend 100 threads when running on Flink because of some scheduling issues where we use to get dead lock in situation similar to the one mentioned by Luke.

I am not sure how it will impact the resource usage if we have 10000 standing threads for single core.

Reply:
also +1 for this change if it does not have performance impact as it will solve a problem which we discussed over the mailing thread https://lists.apache.org/thread.html/0cbf73d696e0d3a5bb8e93618ac9d6bb81daecf2c9c8e11ee220c8ae@%3Cdev.beam.apache.org%3E

Reply:
Yes, this is about avoiding deadlocks (there may be dependencies among the scheduled items that are not reflected in the order that they are received/processed) than performance.